File: spec/sort_spec.js
var bubbleSort = require('../sort.js');

describe('Bubble Sort', function () {

    it('should sort an array of numbers', function () {
      expect(bubbleSort([66,22,67,34])).toEqual([22,34,66,67]);
    });
    it('should return an array of a single number', function () {
      expect(bubbleSort([1])).toEqual([1]);
    });
});

File: spec/sort_spec.rb
require 'spec_helper'
require_relative '../lib/sort'

describe "bubble_sort" do
  it "sorts the array with a simple order" do
      expect([5,3,2,1,4].bubble_sort).to eq([1,2,3,4,5])
  end

  it "sorts the array with negative numbers" do
      expect([-5,3,-2,1,-4].bubble_sort).to eq([-5,-4,-2,1,3])
  end

  it "sorts the array with floats" do
      expect([1.5,3.4,2,1,4].bubble_sort).to eq([1,1.5,2,3.4,4])
  end

  it "sorts the array with a randomized order" do
      expect((1..10).to_a.shuffle.bubble_sort).to eq((1..10).to_a)
  end
end
